lib/syskit/task_context.rb

```diff
@@ -896,6 +914,12 @@ def prepare_for_setup(promise)
 
         # (see Component#perform_setup)
         def perform_setup(promise)
+            unless needs_setup?
```

Comment:
The goal is to never be entering `perform_setup`

I was thinking that the simplest way would be to return true in `setup?` for readonly tasks (i.e. pretend they are set up)

Reply:
Well, I was thinking about using `perform_setup` just for calling `setup_successful!` as otherwise calling `task.start!` would raise an exception.

The same applies for `orocos_task.configure(false)`, which I need in order to transition from `: PRE_OPERATIONAL` to `:STOP` (and then `task.start!` would emit the start_event).